### PR TITLE
ls accessor

### PR DESCRIPTION
**Pull request by dlovell**

### CODE REVIEW

File: python/letsql/backends/let/hotfix.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    @property
+    def op(self):
+        return self.expr.op()
+
+    @property
+    def cached_nodes(self):
+        return self.op.find(CachedNode)
+
+    @property
+    def storage(self):
```
Comment (mesejo) [inline python/letsql/backends/let/hotfix.py:37]: How would this work with multiple storages?

File: python/letsql/backends/let/hotfix.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    def uncached_one(self):
+        if self.is_cached:
+            return self.op.parent.to_expr()
+        else:
+            return self.expr
+
+    def get_key(self):
+        if self.is_cached:
+            return self.storage.get_key(
+                self.ls_con._register_and_transform_cache_tables(
```
Comment (mesejo) [inline python/letsql/backends/let/hotfix.py:128]: Why the use of `_register_and_transform_cache_tables` in here?

**Review by mesejo — APPROVED** [review]

File: python/letsql/backends/let/hotfix.py (hunk 2 above)
Comment (dlovell) [reply python/letsql/backends/let/hotfix.py:128]: currently, the hash depends on all downstream cached nodes being replaced by their cached location

File: python/letsql/backends/let/hotfix.py (hunk 1 above)
Comment (dlovell) [reply python/letsql/backends/let/hotfix.py:37]: `storage` works like `is_cached` and only refers to the top level expr/op being referenced; `storages` / `has_cached` refers to the entire graph of this expr

**Pull request merged**